### ISSUE TITLE
gdal2xyz.py invalid Real value error when specifying `-srcnodata` or `-dstnodata`

### ISSUE THREAD

**Issue by lpinner** [OSGeo/gdal#7410]
## Expected behavior and actual behavior.

gdal2xyz.py fails to export a raster file to xyz when specifying `-srcnodata` or `-dstnodata`. Possibly an argparse issue similar to #5984

## Steps to reproduce the problem.

```
gdal2xyz.py -band 1 -dstnodata -999.0 https://github.com/OSGeo/gdal/raw/master/autotest/gdrivers/data/float32.tif output.txt
usage: <snip>
gdal2xyz.py: error: argument -dstnodata: invalid Real value: '-999.0'

gdal2xyz.py -band 1 -srcnodata -999.0 https://github.com/OSGeo/gdal/raw/master/autotest/gdrivers/data/float32.tif output.txt
usage: <snip>
gdal2xyz.py: error: argument -srcnodata/-nodatavalue: invalid Real value: '-999.0'
```

## Operating system

Ubuntu 22.04 64 bit

## GDAL version and provenance

GDAL 3.6.2, released 2023/01/02 from conda-forge

**Comment by lpinner** [comment]
@rouault I'm preparing a PR for QGIS to get these arguments into the QGIS GDAL2XYZ processing tool and would like to implement a version check before enabling them.  Is it safe to assume this fix will make it into GDAL 3.6.4 (given the backport commit 
2c5315e) or would I be better testing for GDAL 3.7?

PS thanks for this remarkably quick fix.

**Comment by rouault** [comment]
> GDAL 3.6.4

not sure there will be one, but you can safely test for GDAL > 3.6.3 . That will cath both a potential 3.6.4 and 3.7.0
